### PR TITLE
Instructions on incrementing and tagging a version

### PR DESCRIPTION
**Pull request by mcgibbon**
Added a RELEASE.rst file with release instructions. For us right now that means incrementing the version, tagging a commit, and pushing it to master. Further down the line this file can include instructions on deploying to pypi.

### CODE REVIEW

File: RELEASE.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Release Instructions
+====================
+
+1. Prepare master branch for release (make sure all PRs are merged and tests pass).
+
+2. Run `bumpversion <major/minor/bugfix>`. This will create a new tagged commit,
```
Comment (oliverwm1) [inline RELEASE.rst]: Is `bumpversion` a standard tool? We should probably have some info about where to install it from.

File: RELEASE.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Release Instructions
+====================
+
+1. Prepare master branch for release (make sure all PRs are merged and tests pass).
```
Comment (oliverwm1) [inline RELEASE.rst]: Does every merge to master mean an increase in minor version number? How do we decide when to increase version number? (This is probably a better discussion for offline, but maybe we could add some guidelines to our code review document?)

**Comment by oliverwm1** [comment]
My main comment is that it would be helpful to have a link or some small explanation of where to get `bumpversion` from. I'm not familiar with this tool, don't have it installed on my VM or Mac. Also, have a more open-ended comment about versioning, can be discussed more offline.

File: RELEASE.rst (hunk 1 above)
Comment (mcgibbon) [reply RELEASE.rst]: It's on pypi, and it's in the requirements_dev.txt. Package name is bump2version, it behaves the same as the now unsupported bumpversion package.

File: RELEASE.rst (hunk 2 above)
Comment (mcgibbon) [reply RELEASE.rst]: No, you have to manually check out the master branch and perform this release process to increase the version number. I think it's a separate thing from code reviews, but we could have more discussion about our overall code design process. If it's plugged into our current process, we'd discuss this in the morning or during our weekly meetings.

**Comment by mcgibbon** [comment]
Tests now pass on fv3gfs-python, so this is ready to merge when you approve it.

File: RELEASE.rst (hunk 1 above)
Comment (oliverwm1) [reply RELEASE.rst]: Okay sounds good, didn't realize it was in requirements_dev.txt

**Pull request merged**